### PR TITLE
Replace rather than insert into push db on duplicate incoming.

### LINKED ISSUE #2287: notification unread message when locking textsecure
Evertytime I lock TextSecure I get a notification about an unread message from a certain contact (always the same contact), but it is wrong as I already have read all messages.

### PR DESCRIPTION
**Pull request by moxie0**
Combined with the switch to server acked messages, this will
prevent the race condition that occurred when an incoming message
showed up at exactly the moment the app updated.

Fixes #2287

// FREEBIE

### CODE REVIEW

**Comment by McLoo** [comment]
what about a wiping old pending messages?

**Comment by moxie0** [comment]
@McLoo That should do the trick

File: src/org/thoughtcrime/securesms/DatabaseUpgradeActivity.java [hunk 1]
```diff
@@ -58,7 +58,7 @@
   public static final int ASYMMETRIC_MASTER_SECRET_FIX_VERSION = 73;
   public static final int NO_V1_VERSION                        = 83;
   public static final int SIGNED_PREKEY_VERSION                = 83;
-  public static final int NO_DECRYPT_QUEUE_VERSION             = 84;
+  public static final int NO_DECRYPT_QUEUE_VERSION             = 113;
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/DatabaseUpgradeActivity.java]: why 113?

File: src/org/thoughtcrime/securesms/database/PushDatabase.java [hunk 2]
```diff
@@ -80,6 +88,31 @@ public Reader readerFor(Cursor cursor) {
     return new Reader(cursor);
   }
 
+  private Optional<Long> contains(TextSecureEnvelope envelope) {
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/database/PushDatabase.java]: the name "contains" suggests a boolean return to me. more like `find`?

File: src/org/thoughtcrime/securesms/database/PushDatabase.java [hunk 3]
```diff
@@ -31,14 +33,20 @@ public PushDatabase(Context context, SQLiteOpenHelper databaseHelper) {
   }
 
   public long insert(TextSecureEnvelope envelope) {
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/database/PushDatabase.java]: non-null?

**Pull request closed without merging**